x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/components/query_bar/index.tsx

```diff
@@ -168,6 +182,38 @@ export const QueryBarDefineRule = ({
     [field.value]
   );
 
+  const onCloseTimelineModal = useCallback(() => {
+    setLoadingTimeline(true);
+    onCloseTimelineSearch();
+  }, []);
```

Comment:
Additional deps -- `onCloseTimelineSearch` is coming from component props
```suggestion
  }, [onCloseTimelineSearch]);
```